agent_api_rest/src/verification/error.rs

```diff
@@ -0,0 +1,68 @@
+use crate::error::IntoApiErrorExt;
+use crate::DOCUMENTATION_URL;
+use agent_verification::authorization_request::error::AuthorizationRequestError;
+use http_api_problem::ApiError;
+use hyper::StatusCode;
+
+impl IntoApiErrorExt for AuthorizationRequestError {
+    fn into_api_error(self) -> ApiError {
+        use AuthorizationRequestError::*;
+
+        match self {
+            // UniCore API Problem Details
+            AuthorizationRequestBuilderError(_) => ApiError::builder(StatusCode::INTERNAL_SERVER_ERROR)
+                .title("Unexpected Error")
+                .type_url(format!(
+                    "{DOCUMENTATION_URL}problem-details/unexpected#unexpected-error"
+                ))
+                .source(self)
+                .finish(),
+            MissingAuthorizationRequest => ApiError::builder(StatusCode::BAD_REQUEST)
```

Comment:
Should a "Bad Request" really be an "Unexpected Error"? I would go for one of the two.

Reply:
changed it to status code 500